File: Week7/Code/Nets.R
## Build a network

# Author: Rachel Bates (r.bates18@imperial.ac.uk)
# Version: 0.0.1

## Clear the directory ##

rm(list=ls())

## Packages ##

library(igraph)

## Load Data ##

links <- read.csv("../Data/QMEE_Net_Mat_edges.csv", header=T, as.is=T)
nodes <- read.csv("../Data/QMEE_Net_Mat_nodes.csv", header=T, row.names = 1)

###############

#Create graph object
net <- graph.adjacency(as.matrix(links), mode = "directed", weighted=TRUE, diag=F)
        
# Generate colors based on partner type:
colrs <- c("green", "red", "blue")
V(net)$color <- colrs[nodes$Type]

V(net)$size <- 50

# Set edge width based on weight (PhD Students):
E(net)$width <- E(net)$weight

#change arrow size and edge color:
E(net)$arrow.size <- 1
E(net)$edge.color <- "gray80"

E(net)$width <- 1+E(net)$weight/10

graphics.off()

svg("../Output/QMEENet.svg",width=7,height=7)

plot(net, edge.curved=0, vertex.label.color="black") 

legend(x=-1.5, y=-0.1, c("Hosting Partner", "Non-hosting Partner","University"), pch=21,
       col="#777777", pt.bg=colrs, pt.cex=2, cex=.8, bty="n", ncol=1)

dev.off()

File: Week7/Code/Nets.py
#!/usr/bin/env python3

""" Builds a network of links between institutions. Translated from an R script. """

__author__ = 'Rachel Bates r.bates18@imperial.ac.uk'
__version__ = '0.0.1'


## IMPORTS ##

import networkx as nx
import scipy as sc 
import matplotlib.pyplot as p 
import pandas as pd

## CONSTANTS ##

# Reads in the files as pandas dfs
links = pd.read_csv("../Data/QMEE_Net_Mat_edges.csv", header=0)
nodes = pd.read_csv("../Data/QMEE_Net_Mat_nodes.csv", header=0)

################

p.close("all") #Close any open graphics

################

# Makes the column names the same as the row names
links.columns = list(range(0,6))

w_links = list() #initialise list

# Creates a list of tuples, of position and weight of link
for i in range(0,6):
    for j in range(0,6):
        if links[i][j]>0:
            w_links.append((i,j,links[i][j]))

# Creates a list of colours for plot
color_map=["blue", "blue", "green", "green", "green", "red"]

# Create the network plot
pos = nx.random_layout(nodes.index.values)
G = nx.Graph()
G.add_nodes_from(nodes.index.values)
G.add_weighted_edges_from(w_links, arrows=True)


# Initialise figure
plot = p.figure()

# Draw the plot
nx.draw_networkx(G, pos, node_color = color_map, node_size=5000, with_labels=False)
nx.draw_networkx_labels(G, pos, nodes["id"])

plot.savefig('../Output/pyNets.svg') #Save the output